src/xmipp/bindings/python/python_image.cpp

```diff
@@ -34,9 +34,16 @@
 /*                            Image                         */
 /**************************************************************/
 
+ImageObject* createImageObject() {
+    auto *result = PyObject_New(ImageObject, &ImageType);
+    result->image.release(); // Mind the uninitialized memory
+    return result;
+}
+
 /* Destructor */
 void Image_dealloc(ImageObject* self)
 {
+    self->image.reset(); //Free the data (destructor is not executed)
     Py_TYPE(self)->tp_free((PyObject*)self);
```

Comment:
we can define custom deleter to the unique ptr.
But from the Python documentation, it's not clear to me what's the difference between `tp_free` and `PyObject_Del`.